Add some text under the form header

cvalarida opened the pull request:
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/182

## Testing done
:eyes: 

## Screenshots
![image](https://user-images.githubusercontent.com/12970166/126573034-552a82f7-e835-47ee-9451-ad934f77bab3.png)

Commit:
Add some text under the form header

rtwell commented:
I can't quite tell from the screen grab—I wanna make sure the typography is using our gray-dark color (like the rest of our type)

cvalarida commented:
It is not currently. I put the `vads-u-color--gray-medium` class on it. I'll take it off. :+1:

Commit:
Remove the medium gray color

cvalarida commented:
Updated the font color.
![image](https://user-images.githubusercontent.com/12970166/126671780-c976e31c-8901-42a3-8267-6e48612aa5a2.png)

rtwell reviewed (approved):
lgtm!

1Copenut reviewed (approved):
LGTM! I spot checked the branch on `localhost` with MacOS Safari + VoiceOver. The heading and span text was read as expected. We should do a brief spot check in Staging, but won't need more than 5-10 minutes.

Mottie committed:
Add application ID to form header

Mottie commented:
Updated to add the application ID to the header

![Screen Shot 2021-08-10 at 2 00 49 PM](https://user-images.githubusercontent.com/136959/128921596-3dbb9083-74a7-4121-b688-5da96a3569b1.png)

It matches the ID in the green box below the back & continue button

Mottie commented:
Oops, the "Your application will be saved..." will show when you're not logged in & filling out a form that doesn't need authentication, e.g. https://staging.va.gov/careers-employment/education-and-career-counseling/apply-career-guidance-form-28-8832/introduction

Will update to check the logged in state.

Mottie committed:
Only show auto-save message when logged in

Commit:
Add a period

cvalarida commented:
Thanks for the addition, @Mottie! :raised_hands: I just added a period to the end of the sentence, but this looks great. Checking for `isLoggedIn` is perfect. I didn't consider not showing that message to users who aren't logged in, but what you've got there is correct. Because we won't save the data for unauthenticated users. :+1:

Mottie committed:
Merge branch 'master' into form-step-header-a11y-enhancement-cv

Commit:
Merge branch 'master' into form-step-header-a11y-enhancement-cv

Commit:
Merge branch 'form-step-header-a11y-enhancement-cv' of github.com:department-of-veterans-affairs/vets-website into form-step-header-a11y-enhancement-cv

Commit:
Move the . to inside the string

Commit:
Change the pensions test "save" click selector

Because the header now contains the text "save", it wasn't specific
enough. By specifying that the text is only "Save", the test passes.

cvalarida commented on src/applications/pensions/tests/pensions.cypress.spec.js:
```diff
@@ -14,7 +14,7 @@ const fillAdditionalSources = ({ additionalSources }) => {
   cy.findByLabelText(/source/i).type(name);
   cy.findByLabelText(/amount/i).type(amount);
-  cy.findByText(/save/i).click();
+  cy.findByText(/^Save$/).click();
```
The heading now contains the text `Your application will be saved on every change.`, so this selector matched both the heading _and_ the button it should match. Making this selector more specific finds only the button again.

cvalarida commented:
Thanks, @theodur! :raised_hands:
